typed/remove.go

```diff
@@ -58,6 +60,9 @@ func (w *removingWalker) doList(t *schema.List) (errs ValidationErrors) {
 	defer w.allocator.Free(l)
 	// If list is null or empty just return
 	if l == nil || l.Length() == 0 {
+		if REMOVEKEEPEMPTYCOLLECTIONS {
```

Comment:
Should we treat separately empty and nil here? Is that part of the problem?

Reply:
This change wasn't directly part of the problem I was trying to solve but I think something like this will be needed to have a consistent solution. The behavior of this right now always results in a `nil` list even if you specify `[]`